src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/Interop.cs

```diff
@@ -0,0 +1,451 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls.Internal.OpenSsl
+{
+    internal unsafe static class Interop
+    {
+        public static readonly bool IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        public static readonly bool IsOsx = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+
+        private static class OsxLib
+        {
+            public const string SslDll = "libssl.1.0.0.dylib";
```

Comment:
On macOS/OSX we are moving away from OpenSSL.  This would have to be changed to use Apple SecureTransport before it could be used in primetime.